rir/src/compiler/util/type_test.h

```diff
@@ -33,9 +33,10 @@ class TypeTest {
 
         assert(feedback.origin);
         // First try to refine the type
-        if (expected.noAttribs().isA(RType::integer) ||
-            expected.noAttribs().isA(RType::real) ||
-            expected.noAttribs().isA(RType::logical)) {
+        if (!expected.maybeObj() && // TODO: Is this right?
```

Comment:
I think currently noAttrib implies noObj, but it's more robust, so I think it's ok.